src/shared_utilities/FileReader.jl

```diff
@@ -392,32 +440,78 @@ return the same value unless `date_idx` is modified elsewhere in between.
 - DateTime or DateTimeNoLeap
 """
 next_date_in_file(prescribed_data::PrescribedDataTemporal) =
-    prescribed_data.file_info.all_dates[prescribed_data.file_state.date_idx[1] + Int(
-        1,
-    )]
+    prescribed_data.file_info.all_dates[first(
+        prescribed_data.file_states,
+    )[2].date_idx[1] + Int(1)]
 
 """
-    interpolate_data(prescribed_data::PrescribedDataTemporal, date::Union{DateTime, DateTimeNoLeap}, space::Spaces.AbstractSpace)
+    get_data_at_date(
+        prescribed_data::PrescribedDataStatic,
+        space::Spaces.AbstractSpace,
+        varname::String,
+    )
 
-Interpolates linearly between two `Fields` in the `prescribed_data` struct.
+Returns the data at a given date, interpolated if necessary.
 
 # Arguments
 - `prescribed_data`      # contains fields to be interpolated.
-- `date`                 # start date for data.
 - `space`                # the space of our simulation.
+- `varname`              # the name of the variable we want to read in.
 
 # Returns
 - Fields.field
 """
-function interpolate_data(
+function get_data_at_date(
+    prescribed_data::PrescribedDataStatic,
+    space::Spaces.AbstractSpace,
+    varname::String,
+)
+    (; regrid_dirpath, outfile_root) = prescribed_data.file_info
+
+    comms_ctx = ClimaComms.context(space)
+    field = Regridder.read_from_hdf5(
+        regrid_dirpath,
+        outfile_root,
+        nothing,
+        varname,
+        comms_ctx,
+    )
+    Regridder.swap_space!(field, space)
+    return nans_to_zero.(field)
```

Comment:
Probably longer term this shouldnt be in these functions, and we should assume the input data has been cleaned and is ready for use already.

Reply:
(I think this specific replacement may be correct for albedo, for example, but not other fields)